src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -35,6 +35,16 @@ namespace NUnit.Engine.Services
     /// </summary>
     public class DefaultTestRunnerFactory : InProcessTestRunnerFactory, ITestRunnerFactory
     {
+        private IProjectService _projectService;
+
+        public override void StartService()
+        {
+            _projectService = ServiceContext.GetService<IProjectService>();
+            Status = _projectService != null && ((IService)_projectService).Status == ServiceStatus.Started
```

Comment:
Yes, I came across this anomaly while I was working on it. I briefly  tried making each interface derive from IService, but I ran into problems and "resolved" the issue using a cast.

The reason it's not so simple is that services are used by both clients (runners) and internally by the engine. The interface that returns them to clients returns either an object or the requested type, depending on whether or not the generic call is used. IService is an internal interface, since clients can't initialize services, etc., so that can't be returned externally.

Now that I've been away from the code for a day, I'll take another look.